src/main/java/org/minions/devfund/mauricioz/WhackAMole.java

```diff
@@ -0,0 +1,161 @@
+package org.minions.devfund.mauricioz;
+
+/**
+ * class in charge to provide game functionality.
+ */
+public class WhackAMole {
+    private static final int MOLES_LEFT = 10;
+    private int score = 0;
+    private int molesLeft;
+    private int attemptsLeft;
+    private char[][] moleGrid;
+
+    /**
+     * constructor.
+     * @param numAttempts number of attempts for the game.
+     * @param gridDimension grid size.
+     */
+    public WhackAMole(int numAttempts, int gridDimension) {
+        attemptsLeft = numAttempts;
+        moleGrid = new char[gridDimension][gridDimension];
+        this.molesLeft = MOLES_LEFT;
+        initializeMoleGrid();
+        placeMolesInTheGrid();
+    }
+
+    /**
+     * method to verify if whack was over a mole or not.
+     * @param x coordinate in x.
+     * @param y coordinate in y.
+     * @return true if mole was whacked.
+     */
+    public boolean place(int x, int y) {
+        if (moleGrid[x][y] == 'M') {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)